lxd/request/request.go

```diff
@@ -78,7 +78,10 @@ func CountStartedRequest(r *http.Request) {
 // The callback function is used to mark the request as completed for the API metrics.
 func MetricsCallback(request *http.Request, result metrics.RequestResult) {
 	callback, err := GetCtxValue[func(metrics.RequestResult)](request.Context(), MetricsCallbackFunc)
-	if err != nil && (strings.HasPrefix(request.URL.Path, "/1.0") || request.URL.Path == "/") {
+
+	// Verify the auth method in the request context to determine if the request comes from the /dev/lxd socket.
+	authMethod, _ := GetCtxValue[string](request.Context(), CtxProtocol)
+	if err != nil && strings.HasPrefix(request.URL.Path, "/1.0") && authMethod != api.AuthenticationMethodDevLxd {
```

Comment:
Once the authentication method is moved to `lxd/auth/types.go` the import cycle will be unavoidable here. I suggest moving the callback to `lxd/metrics.go`